### DISCUSSION TITLE
Synching Production to Local Dev

### DISCUSSION THREAD

**Discussion by Digital-Pig-LLC** [markshust/docker-magento#1261] [Q&A]
I'm looking for some advice for the given title.

The contract I'm just starting on has a couple of Magento instances for prod and staging. I've used the work Mark has done here to create a local docker environment (PHP 8.2, Magento 2.4.2 (2.4.7 on staging site)), and I'd like to start getting it in sync with the client's site.
It's been a number of years since I dealt with Magento. I wasn't given the contract as a Magento dev, but as a PHP engineer, as they have other non-Magento issues.
So, to that end, I'm wondering if anyone here has done something similar and has any thoughts on the least painful way to go about this.

Any tips, tricks, thought, ideas, gotchas, etc. would be welcome. I'm not opposed to scripting this either, as I imagine there's no single route to success, but probably lots of exporting etc., installing stuff the previous dev installed (Bss_AjaxCart is one that's broken)

Thank you kindly.
Paul

**Comment by its-leofisher** [comment]
Definitely a lot to consider but here is a quick get it up and running sync.

These are typically the steps I perform on a general level:
- On their production server I run n98-magerun2 db:dump with Strip arguments particularly Customer and Order data so there is no worry towards data leaks on your side, [see here](https://github.com/netz98/n98-magerun2?tab=readme-ov-file#stripped-database-dump)
- Go through [Marks Existing site Install process](https://github.com/markshust/docker-magento?tab=readme-ov-file#existing-projects)
   - Load their source code into /src folder
   - Update .env file
   - Update configs matching their staging website, hopefully they are using Sandbox APIs and Staging settings.  Update either with commands or via Admin if you can get in at this stage.
   - Make sure I can run xDebug
   - Connect to the DB via GUI like Workbench or Tableplus

When the site loads, I then QA everything locally starting with the critical path while tailing debug.log and exceptions.log:

Critical Path Test Scenarios:
 - Home > Category > PDP > Add To Cart > Cart > Checkout. 
     - Then repeat that for different product scenarios depending on the complexity of their inventory as a Guest and a Customer.
     - This will also give you test data for future work
 - Validate My Account Actions
 - Explore their Content Management Process
 - Test external services

Hope that helps, there is probably more involved but I can help if you need it!

**Reply by Digital-Pig-LLC** [reply]
Awesome, sir! Thanks for the input, that will give me a good start. 

I'm already about 80% through a Customer class in Python which uses Selenium to pull up a browser, log in, pick a random category, then from there pick some random products and add them to the cart. Then through the checkout process. This will allow me to capture some metrics on performance, etc. and stash them to a database to report on. 
I can then run that on a schedule and also add it to a git hook as a post-deploy test to make sure I haven't broken anything.

Thanks again!
